app/src/backend/environment/hooks/useEnvironmentManager.ts

```diff
@@ -61,23 +61,13 @@ const useEnvironmentManager = (initListenerAndFetcher: boolean = false) => {
       setIsLoading(true);
       fetchAllEnvironmentDetails(ownerId)
         .then((environmentMap) => {
-          dispatch(environmentVariablesActions.setAllEnvironmentData({ environmentMap }));
-
-          // if there are no environments, create a default one
-          if (Object.keys(environmentMap).length === 0) {
-            addNewEnvironment("Default").then((defaultEnv) => {
-              if (defaultEnv) {
-                setCurrentEnvironment(defaultEnv.id);
-              }
-            });
-          } else if (!currentEnvironmentId) {
-            // if there is no active environment, set the first environment as the active environment
-            const defaultEnvironment = Object.keys(environmentMap)[0];
-            setCurrentEnvironment(defaultEnvironment);
+          if (Object.keys(environmentMap).length > 0 && !environmentMap[currentEnvironmentId]) {
+            // setting the first environment as the current environment if the current environment is not found in environmentMap
```

Comment:
When will this case happen?

Reply:
switching workspaces and auth state changes